crux_core/src/lib.rs

```diff
@@ -170,12 +170,12 @@ pub use self::{
 /// as the type argument to [`Core`] or [`Bridge`](bridge::Bridge).
 pub trait App: Default {
     /// Event, typically an `enum`, defines the actions that can be taken to update the application state.
-    type Event: Send + 'static;
+    type Event: Send + Deserialize<'static> + 'static;
```

Comment:
Might be nice to avoid adding the `Deserialize` bound here.  If you're doing a pure-rust `App` you wouldn't need to go through serde and you wouldn't need the typegen either.  This requirement could be lightly annoying for those cases.  

Could maybe put these bounds on `TypeGenregister::app` instead?

Reply:
Great spot @obmarg ! Fixed